src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -30712,20 +30712,567 @@ public interface I1
+            var cctor = i1.GetMember<MethodSymbol>(".cctor");
+
+            Assert.False(cctor.IsAbstract);
+            Assert.False(cctor.IsVirtual);
+            Assert.False(cctor.IsMetadataVirtual());
+            Assert.False(cctor.IsSealed);
+            Assert.True(cctor.IsStatic);
+            Assert.False(cctor.IsExtern);
+            Assert.False(cctor.IsAsync);
+            Assert.False(cctor.IsOverride);
+            Assert.Equal(Accessibility.Private, cctor.DeclaredAccessibility);
+            Assert.Equal(MethodKind.StaticConstructor, cctor.MethodKind);
+        }
+
+        [Fact]
+        public void Constructors_05()
+        {
+            var source1 =
+@"
+interface I1
+{
+    static string F = ""F"";
+
+    static void Main() 
+    {
+        System.Console.WriteLine(F);
+        System.Console.WriteLine(""Main"");
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                                                 parseOptions: TestOptions.Regular);
+
+            compilation1.VerifyDiagnostics();
+
+            ValidateConstructor(compilation1.SourceModule);
+
+            CompileAndVerify(compilation1, symbolValidator: ValidateConstructor, expectedOutput:
+@"F
+Main
+");
+        }
+
+        [Fact]
+        public void Constructors_06()
+        {
+            var source1 =
+@"
+interface I1
+{
+    static string F = ""F"";
+
+    static I1()
+    {
+        System.Console.WriteLine(F);
+        System.Console.WriteLine(""I1"");
+    }
+
+    static void Main() 
+    {
+        System.Console.WriteLine(""Main"");
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe.WithMetadataImportOptions(MetadataImportOptions.All),
+                                                 parseOptions: TestOptions.Regular);
+
+            compilation1.VerifyDiagnostics();
+
+            ValidateConstructor(compilation1.SourceModule);
+
+            CompileAndVerify(compilation1, symbolValidator: ValidateConstructor, expectedOutput:
+@"F
+I1
+Main
+");
+        }
+
+        [Fact]
+        public void Constructors_07()
+        {
+            var source1 =
+@"
+interface I1
+{
+    extern static I1();
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                                                 parseOptions: TestOptions.Regular);
+
+            compilation1.VerifyDiagnostics(
+                // (4,19): warning CS0824: Constructor 'I1.I1()' is marked external
+                //     extern static I1();
+                Diagnostic(ErrorCode.WRN_ExternCtorNoImplementation, "I1").WithArguments("I1.I1()").WithLocation(4, 19)
                 );
+
+            validate(compilation1.SourceModule);
+
+            void validate(ModuleSymbol m)
```

Comment:
Minor nit: local method seems unnecessary. If you want to keep it, please move to the end of the function.